spec/models/user_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe '#name_or_email' do
+    context 'when a name is registered' do
+      it 'returns a name' do
+        alice = create(:user, :alice)
+        expect('アリス').to eq alice.name_or_email
+      end
+    end
+
+    context 'when a name is not registered' do
+      it 'returns a email' do
+        nameless = create(:user, :nameless)
```

Comment:
```suggestion
        nameless = create(:user, name: "")
```

traitにしちゃうと、「nameが設定されていない」という事実ををFactoryの定義を見に行かないと確認できないので、あえてtraitにしない方がわかりやすいです。
（`nameless`という名前から想像は付きやすいが、それでも上のコードの方が見て明らか）

Reply:
そうですね。改善されたコードだと`trait`の定義を見に行かなくて済むのでよいですね！！
こちら修正しました。